scripts/get-git-tag-name.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# This script derives a git tag name from the version fields found in a given Go
+# file. It also checks if the derived git tag name is a valid SemVer compliant
+# version string.
+
+# get_git_tag_name reads the version fields from the given file and then
+# constructs and returns a git tag name.
+get_git_tag_name() {
+  local file_path="$1"
+
+  # Check if the file exists
+  if [ ! -f "$file_path" ]; then
+      echo "Error: File not found at $file_path" >&2
+      exit 1
+  fi
+
+  # Read and parse the version fields.
+  local app_major=$(grep -oP 'AppMajor\s*uint\s*=\s*\K\d+' "$file_path")
```

Comment:
The variables could be auto-formatted as a block:
```
	AppMajor  uint = 0
	AppMinor  uint = 3
	AppPatch  uint = 2
	AppStatus      = "alpha"
```

which adds a variable number of white spaces everywhere except for after the `=`.

Broadly, my idea here was to try to make the matching as robust as possible. With this idea of matching any legitimate golang variable deceleration of a given name, rather than any particular formatting.

Should I make the matching more specific?

Reply:
Mm that makes sense, didn't think of these spacing changes `gofmt` sometimes makes.